.scripts/check-dist.mjs

```diff
@@ -0,0 +1,77 @@
+import { deepStrictEqual, doesNotThrow } from 'node:assert';
+import path from 'node:path';
+import { describe, test } from 'node:test';
+import { createRequire } from 'node:module';
+
+const rootDir = path.resolve(import.meta.dirname, '..');
+const require = createRequire(import.meta.url);
+const { publishConfig } = require('../package.json');
+
+const entrypoints = getEntrypoints(rootDir, publishConfig.exports);
+
+test('all entrypoints are configured', () => {
+  deepStrictEqual(
+    entrypoints.map(entry => entry.name),
+    [
+      'es-toolkit',
+      'es-toolkit/array',
+      'es-toolkit/compat',
+      'es-toolkit/function',
+      'es-toolkit/math',
+      'es-toolkit/object',
+      'es-toolkit/predicate',
+      'es-toolkit/promise',
+      'es-toolkit/string',
```

Comment:
Fixed list of entrypoints that should be defined.

This is meant to prevent issues where we accidentally drop some values in `package.json#publishConfig#exports`.

Reply:
This looks great, while I think we could configure this using vitest, not Node.js' native test module. (It would be great if we could test this altogether with other unit tests.)

Let me modify this a little bit with vitest after merging this.

Reply:
One possible issue with Vitest is that it's using Vite, which is built to be more tolerant of things like importing CJS sources as ESM modules, or ESM sources in a CJS context.

Using Node's testing capabilities is a bit "closer to the metal", which can be useful for this specific purpose.

But there's value in using Vitest for all the things too.

Just something to keep in mind.